### PR TITLE
test: using minikube profiles for rook cluster creation

### PR DESCRIPTION
**Pull request by rkachach**
**Description of your changes:**
Let's use the minikube profiles feature to set up a unique profile exclusively for the Rook cluster. By doing this, we make sure that we don't run into any issues with other minikube profiles that users might already have in their local setup.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by sp98 — APPROVED** [review]

File: tests/scripts/create-dev-cluster.sh [hunk 1]
```diff
@@ -63,24 +64,25 @@ show_ceph_dashboard_info() {
...
     fi
 }
 
 setup_minikube_env() {
     minikube_driver="$(get_minikube_driver)"
     echo "Setting up minikube env (using $minikube_driver driver)"
-    minikube delete
-    minikube start --disk-size=40g --extra-disks=3 --driver "$minikube_driver"
-    eval "$(minikube docker-env -p minikube)"
+    minikube delete --profile $ROOK_PROFILE_NAME
```
Comment (BlaineEXE) [inline tests/scripts/create-dev-cluster.sh:83]: In general, when writing scripts, it's best for other users (anyone reading the script now or in the future, with or without the intent to modify it) to use fully-expressed flag names like `--profile` which you are doing here but not where in the script. This is especially important with scripts that are less often used, like this one. I'd suggest using the long form everywhere.

Comment (rkachach) [reply tests/scripts/create-dev-cluster.sh:83]: Fixed

File: tests/scripts/create-dev-cluster.sh [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env bash
 
 
+ROOK_EXAMPLES_DIR="../../deploy/examples/"
```
Comment (obnoxxx) [inline tests/scripts/create-dev-cluster.sh:4]:
```suggestion
ROOK_EXAMPLES_DIR="${SCRIPT_ROOT}/../../deploy/examples/"
```
this suggestion fixes a mistake I made when resolving rebase/merge conflicts.

**Review by obnoxxx — CHANGES_REQUESTED** [review]

**Review by obnoxxx — APPROVED** [review]
LGTM, thanks!

File: tests/scripts/create-dev-cluster.sh [hunk 3]
```diff
@@ -51,24 +53,25 @@ show_ceph_dashboard_info() {
     echo "   API_HOST: $PROMETHEUS_API_HOST"
     fi
     echo "==========================="
+    echo " "
+    echo " *** To start using your rook cluster please set the following alias: "
+    echo " "
+    echo "    > " alias kubectl=\"minikube --profile $ROOK_PROFILE_NAME kubectl --\"
```
Comment (leseb) [inline tests/scripts/create-dev-cluster.sh:59]: Why not print the `KUBECTL` variable here to avoid the rewrite?

**Review by leseb — COMMENTED** [review]
Given that `--profile $ROOK_PROFILE_NAME` is used in most of the `minikube` calls, we might consider using a variable like: `MINIKUBE=minikube --profile $ROOK_PROFILE_NAME` in the future, to add more args to the main command. Mostly good, small nit.

File: tests/scripts/create-dev-cluster.sh (hunk 3 above)
Comment (rkachach) [reply tests/scripts/create-dev-cluster.sh]: @leseb I like the idea of the MINIKUBE variable. I updated the code to introduce the proposed changes.

**Review by leseb — APPROVED** [review]

**Comment by parth-gr** [comment]
@rkachach by setting profile can we start multiple minikube cluster at same time?

**Comment by rkachach** [comment]
> @rkachach by setting profile can we start multiple minikube cluster at same time?

@parth-gr current version doesn't but the PR https://github.com/rook/rook/pull/13173 adds the support :)

**Pull request merged**